androdeb

```diff
@@ -53,14 +54,45 @@ case $key in
 esac
 done
 
-if [ ! -z "$PULL" ]; then
+##########################################################
+#  GIT-PULL
+##########################################################
+
+if [ ! -z "$GIT_PULL" ]; then
 	echo "Updating androdeb by git pull"
 	cd $spath
 	git pull
 	echo "Done."
 	exit 0
 fi
 
+##########################################################
+#  PULL
+##########################################################
+if [ ! -z "$PULL" ]; then
+	if [ $1 == "-a" ]; then
+		PRESERVE="-a"
+		echo "Preserving filestamps and mode"
+		shift || true
+	fi
+
+	if [ "2" -le "$#" ]; then
+		i=0
+		while [ "2" -le "$#" ]; do
+			FILES["$i"]=/data/androdeb/debian/$1
+			shift || true
+			i=$((i + 1))
+		done
+	else
+		FILES["0"]=/data/androdeb/debian/$1
+		shift || true
```

Comment:
why is this shift needed?

Reply:
I need to consume the argument because the last argument (local destination) is passed through below